src/Microsoft.AspNet.Mvc.Core/Formatters/JsonPatchInputFormatter.cs

```diff
@@ -1,126 +1,50 @@
-﻿using System;
-using System.IO;
-using System.Text;
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
 using System.Threading.Tasks;
+using Microsoft.AspNet.JsonPatch;
 using Microsoft.Framework.Internal;
 using Microsoft.Net.Http.Headers;
-using Newtonsoft.Json;
 
 namespace Microsoft.AspNet.Mvc
 {
-    public class JsonPatchInputFormatter : InputFormatter
+    public class JsonPatchInputFormatter : JsonInputFormatter
     {
-        private const int DefaultMaxDepth = 32;
-        private JsonSerializerSettings _jsonSerializerSettings;
-
         public JsonPatchInputFormatter()
         {
-            SupportedEncodings.Add(Encodings.UTF8EncodingWithoutBOM);
-            SupportedEncodings.Add(Encodings.UTF16EncodingLittleEndian);
-
-            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("application/jsonpatch"));
-
-            _jsonSerializerSettings = new JsonSerializerSettings
-            {
-                MissingMemberHandling = MissingMemberHandling.Ignore,
-
-                // Limit the object graph we'll consume to a fixed depth. This prevents stackoverflow exceptions
-                // from deserialization errors that might occur from deeply nested objects.
-                MaxDepth = DefaultMaxDepth,
+            // Remove json mediatype header and only include json-patch+json value.
+            SupportedMediaTypes.Remove(MediaTypeHeaderValue.Parse("application/json"));
+            SupportedMediaTypes.Remove(MediaTypeHeaderValue.Parse("text/json"));
 
-                // Do not change this setting
-                // Setting this to None prevents Json.NET from loading malicious, unsafe, or security-sensitive types
-                TypeNameHandling = TypeNameHandling.None
-            };
-
-            _jsonSerializerSettings.ContractResolver = new JsonContractResolver();
+            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("application/json-patch+json"));
         }
 
-        /// <summary>
-        /// Gets or sets the <see cref="JsonSerializerSettings"/> used to configure the <see cref="JsonSerializer"/>.
-        /// </summary>
-        public JsonSerializerSettings SerializerSettings
+        /// <inheritdoc />
+        public override Task<object> ReadRequestBodyAsync([NotNull] InputFormatterContext context)
         {
-            get { return _jsonSerializerSettings; }
-            set
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException(nameof(value));
-                }
+            var jsonSerializer = CreateJsonSerializer();
+            var jsonPatchDocument = base.ReadRequestBodyAsync(context).Result as IJsonPatchDocument;
```

Comment:
`as` is the wrong case. Only use `as` when you want to have conditional behavior. Use a c-style case if you expect it to always succeed.